index.html

```diff
@@ -811,6 +814,94 @@ <h3>
+          start URL.
+        </p>
+        <p>The [=identity=] is used by user agents to uniquely identify the
+        application universally. When the user agent sees a manifest with an
+        [=identity=] that does not correspond to an already-installed application,
+        it SHOULD treat that manifest as a description of a distinct
+        application, even if it is served from the same URL as that of another
+        application. When the user agent sees a manifest with an [=identity=]
+        matching that of an already-installed application, it SHOULD be used as
+        a signal that this manifest is a replacement for the already-installed
+        application's manifest, and not a distinct application, even if it is
+        served from a different URL than the one seen previously.
+        </p>
+        <p>
+        The [=identity=] is considered matching when the processed |manifest|["id"] [=url/equal|equals=].
+        </p>
+        <p class="note">
+          The [=identity=] can be used by a service that collects lists of web
+          applications to uniquely identify applications.
+        </p>
+        <p>
+          To <dfn>process the `id` member</dfn>, given [=object=]
+          |json:JSON|, [=ordered map=] |manifest:ordered map|:
+        </p>
+        <ol class="algorithm">
+          <li>Set |manifest|["id"] to |manifest|["start_url"].
+          </li>
+          <li>If the type of |json|["id"] is not [=string=], return.
+          </li>
+          <li>Let |id:URL| be the result of [=URL Parser|parsing=] |json|["id"]
+          with |manifest|["start_url"]'s origin as the base URL.
+          </li>
+          <li>If |id| is failure, return.
+          </li>
+          <li>If |id| is not [=same origin=] as |manifest|["start_url"], return.
+          </li>
+          <li>Otherwise, set |manifest|["id"] to |id|.</li>
+        </ol>
+        <aside class="example">
+          <p>Below table shows some example cases of the |identity| processing algorithm.</p>
+          <table class="data">
+            <tr>
+              <th>|json|["id"]</th>
+              <th>|manifest|["start_url"]</th>
+              <th>|manifest|["id"]</th>
+            </tr>
+            <tr>
+              <td><i>undefined</i></td>
+              <td>"https://example.com/resources/start"</td>
+              <td>"https://example.com/resources/start"</td>
+            </tr>
+            <tr>
+              <td>""</td>
+              <td>"https://example.com/resources/start"</td>
+              <td>"https://example.com/"</td>
+            </tr>
+            <tr>
+              <td>"/"</td>
+              <td>"https://example.com/resources/start"</td>
+              <td>"https://example.com/"</td>
+            </tr>
+            <tr>
+              <td>"foo"</td>
+              <td>"https://example.com/resources/start"</td>
+              <td>"https://example.com/foo"</td>
```

Comment:
This one is also incorrect, I believe: 

```suggestion
              <td>"https://example.com/resources/start/foo"</td>
```

Reply:
the app's origin is used as the base url for the url parsing instead of start_url. This makes sure that the id stability doesn't depend on start_url.

Reply:
Oh, whoops! Yes, I misread the algorithm. That's easy to miss, so, I think we should pull origin out into a variable and then use that as the base.